### PR TITLE
fix(datepicker): minValidator & maxValidation false errors

### PR DESCRIPTION
**Pull request by PhilipSultanescu**
minValidator & maxValidation should not give errors when value = min or max

Example setting the [min]="today" and [value]="today" will trigger the mdDatepickerMin validation error, but it shouldn't. Same goes for [max]. It should include as valid also the min and max dates.

Tested with v2.0.0.beta.5

### CODE REVIEW

**Comment by PhilipSultanescu** [comment]
I signed it!

**Comment by PhilipSultanescu** [comment]
I signed it!

**Comment by PhilipSultanescu** [comment]
I signed it!

**Comment by PhilipSultanescu** [comment]
I signed it!

**Comment by jelbourn** [comment]
@mmalerba not sure if you had a reason for having it this way. If we do want this change, it needs unit tests.

**Comment by mmalerba** [comment]
@jelbourn nope, just a mistake.

@PhilipSultanescu mind adding unit tests that would have caught this?

**Comment by PhilipSultanescu** [comment]
Plunker: just focus in and out to trigger the validation.
https://plnkr.co/edit/VtxBbgkwAohgsrsOnmLp?p=preview

@mmalerba, @jelbourn Working now on the unit tests. Will also switch my CLA to Corporate, pending approvals.

**Comment by PhilipSultanescu** [comment]
Tests commited.

File: src/lib/datepicker/datepicker.spec.ts [hunk 1]
```diff
@@ -433,6 +433,58 @@ describe('MdDatepicker', () => {
         expect(testComponent.datepicker._minDate).toEqual(new Date(2010, JAN, 1));
         expect(testComponent.datepicker._maxDate).toEqual(new Date(2020, JAN, 1));
       });
+
+      it('should mark input invalid', async(() => {
```
Comment (mmalerba) [inline src/lib/datepicker/datepicker.spec.ts]: can you split this into separate tests?
```
'should mark invalid when value is before min'
'should mark invalid when value is after max'
'should not mark invalid when value equals min'
'should not mark invalid when value equals max'
'should not mark invalid when value is between min and max'
```
this way we avoid the deep nesting and if we get a failure its easier to see why

Comment (PhilipSultanescu) [inline src/lib/datepicker/datepicker.spec.ts]: tried it but got errors when split, will debug it more tomorrow to find the cause and reply.

**Comment by mmalerba** [comment]
LGTM, but can you squash the commits? The tool we use to merge into Google's codebase doesn't like the merge commits. I think it might also fix the cla error.

**Comment by PhilipSultanescu** [comment]
Closed this request because of too many commit conflicts, will reopen using a feature branch.
Maybe it will also solve the CLA problem.

**Pull request closed without merging**